test/type.t.cpp

```diff
@@ -313,6 +321,14 @@ CASE( "boolean: Allows to default-construct a boolean thus defined" )
     EXPECT( to_value( x ) == false );
 }
 
+CASE( "boolean: Allows to custom-default-construct a boolean thus defined" )
+{
+    typedef boolean< struct Tag, custom_default_t< bool, true > > CustomDefaultType;
```

Comment:
Use `nonstd::boolean` to prevent ambiguity for `boolean`:
```
typedef nonstd::boolean< struct Tag, custom_default_t< bool, true > > CustomDefaultType;
```